### PR TITLE
Bump dekorate from 3.0.0 to 3.0.2

### LINKED ISSUE #26587: Route generated by OpenShift extension should use named port
### Describe the bug

When generating openshift yamls using the OpenShift extension, the `Route` object that gets generated should use named ports. This is especially important if you run the application on a different port than the default (i.e. by setting `quarkus.http.port` to something).

### Expected behavior

I would expect the `Route` to look like this:

```yaml
apiVersion: route.openshift.io/v1
kind: Route
metadata:
  annotations:
    app.openshift.io/vcs-url: <<unknown>>
    app.quarkus.io/build-timestamp: 2022-07-06 - 14:54:14 +0000
  labels:
    app.kubernetes.io/name: route-wrong
    app.kubernetes.io/version: 1.0-SNAPSHOT
    app.openshift.io/runtime: quarkus
  name: route-wrong
spec:
  port:
    targetPort: http
  to:
    kind: Service
    name: route-wrong
```

### Actual behavior

But instead, it looks like this:

```yaml
apiVersion: route.openshift.io/v1
kind: Route
metadata:
  annotations:
    app.openshift.io/vcs-url: <<unknown>>
    app.quarkus.io/build-timestamp: 2022-07-06 - 14:54:14 +0000
  labels:
    app.kubernetes.io/name: route-wrong
    app.kubernetes.io/version: 1.0-SNAPSHOT
    app.openshift.io/runtime: quarkus
  name: route-wrong
spec:
  port:
    targetPort: 8082
  to:
    kind: Service
    name: route-wrong
```

The problem with this is that with the `Service` that is generated, the `Service` accepts requests on port `80` and not `8082`.

```yaml
apiVersion: v1
kind: Service
metadata:
  annotations:
    app.openshift.io/vcs-url: <<unknown>>
    app.quarkus.io/build-timestamp: 2022-07-06 - 14:54:14 +0000
  labels:
    app.kubernetes.io/name: route-wrong
    app.kubernetes.io/version: 1.0-SNAPSHOT
    app.openshift.io/runtime: quarkus
  name: route-wrong
spec:
  ports:
    - name: http
      port: 80
      targetPort: 8082
  selector:
    app.kubernetes.io/name: route-wrong
    app.kubernetes.io/version: 1.0-SNAPSHOT
  type: ClusterIP
```

By changing the `Route` to use the named port (`http`), then the actual ports don't really matter so long as the `Service` is correct (which it is).

### How to Reproduce?

Reproducer:

[route-wrong.zip](https://github.com/quarkusio/quarkus/files/9055904/route-wrong.zip)

1. Unzip reproducer
2. `cd route-wrong`
3. `./mvnw package`
4. Examine `target/kubernetes/openshift.yml`

### Output of `uname -a` or `ver`

```
Darwin edeandrea-m1pro 21.5.0 Darwin Kernel Version 21.5.0: Tue Apr 26 21:08:37 PDT 2022; root:xnu-8020.121.3~4/RELEASE_ARM64_T6000 arm64
```

### Output of `java -version`

```
openjdk version "17.0.3" 2022-04-19
OpenJDK Runtime Environment Temurin-17.0.3+7 (build 17.0.3+7)
OpenJDK 64-Bit Server VM Temurin-17.0.3+7 (build 17.0.3+7, mixed mode)
```

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.10.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

```
Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537)
Maven home: /Users/edeandre/.m2/wrapper/dists/apache-maven-3.8.4-bin/52ccbt68d252mdldqsfsn03jlf/apache-maven-3.8.4
Java version: 17.0.3, vendor: Eclipse Adoptium, runtime: /Users/edeandre/.sdkman/candidates/java/17.0.3-tem
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "12.4", arch: "aarch64", family: "mac"
```

### Additional information

_No response_

### LINKED ISSUE #28078: quarkus-kind extension does not create manifests (kind.yml/.json)
### Describe the bug

When using the `quarkus-kind` extension, no manifests are created and the deployment subsequently fails.

I'm using the following command to build the project: 
`gradle build -Dquarkus.container-image.build=true -Dquarkus.kubernetes.deploy=true`

### Expected behavior

I expect the `quarkus-kind` extension to create a `kind.yml` and `kind.json` file.

### Actual behavior

No files are created. The deployment fails.

### How to Reproduce?

[crickets.zip](https://github.com/quarkusio/quarkus/files/9605066/crickets.zip)

Use the following command: `gradle build -Dquarkus.container-image.build=true -Dquarkus.kubernetes.deploy=true`

### Output of `uname -a` or `ver`

Darwin V6624MV3PM 21.6.0 Darwin Kernel Version 21.6.0: Mon Aug 22 20:19:52 PDT 2022; root:xnu-8020.140.49~2/RELEASE_ARM64_T6000 arm64

### Output of `java -version`

openjdk 17.0.4 2022-07-19

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.11.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

------------------------------------------------------------ Gradle 7.4.2 ------------------------------------------------------------  Build time:   2022-03-31 15:25:29 UTC Revision:     540473b8118064efcc264694cbcaa4b677f61041  Kotlin:       1.5.31 Groovy:       3.0.9 Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021 JVM:          17.0.4 (Eclipse Adoptium 17.0.4+8) OS:           Mac OS X 12.6 aarch64

### Additional information

_No response_

### LINKED ISSUE #28297: Default Service Ports Protocol Cannot be Added
## Bug
The generated kubernetes.yml `Service` > `spec` > `ports` > `protocol` cannot be accessed using the following application.properties configuration:  `quarkus.kubernetes.ports."http".protocol=tcp`.

Using `quarkus.kubernetes.ports."http".container-port` and `quarkus.kubernetes.ports."http".host-port` as per the documentation, I am able to change the default kubernetes.yml `Service` > `spec` > `ports` > `targetPort` and the default kubernetes.yml `Service` > `spec` > `ports` > `port` respectively. I mention this to reference the syntax - I used the results of these latter configurations to confirm the syntax for the former is not the issue.

**application.properties**
```
quarkus.kubernetes.add-build-timestamp=false
quarkus.kubernetes.add-version-to-label-selectors=false
quarkus.kubernetes.prometheus.annotations=false

#  Append Ingress component.
quarkus.kubernetes.ingress.expose=true

#  metadata > name
quarkus.kubernetes.name=name

#  metadata > namespace
quarkus.kubernetes.namespace=namespace

quarkus.kubernetes.ports."http".protocol=tcp
quarkus.kubernetes.ports."http".container-port=8080
quarkus.kubernetes.ports."http".host-port=8080
```

**kubernetes.yml (service portion)**
```
apiVersion: v1
kind: Service
metadata:
  annotations:
    app.quarkus.io/commit-id: 5c4720d733cd2bd9bea27be621104d38e4609910
  labels:
    app.kubernetes.io/name: name
  name: name
  namespace: namespace
spec:
  ports:
    - name: http
      port: 8080
      targetPort: 8080
  selector:
    app.kubernetes.io/name: name
  type: ClusterIP
```

**Expected kubernetes.yml (service portion)**
```
apiVersion: v1
kind: Service
metadata:
  annotations:
    app.quarkus.io/commit-id: 5c4720d733cd2bd9bea27be621104d38e4609910
  labels:
    app.kubernetes.io/name: name
  name: name
  namespace: namespace
spec:
  ports:
    - name: http
      port: 8080
      targetPort: 8080
      protocol: TCP            # Here
  selector:
    app.kubernetes.io/name: name
  type: ClusterIP
```

## To Repoduce

1. Generate a Quarkus project
2. Add the extension quarkus-kubernetes
3. Add the application.properties file noted above
4. Run the project

**Quarkus Version**
```
<quarkus.platform.version>2.11.2.Final</quarkus.platform.version>
```

_On a side note:_
Is there any way to remove the default `annotations` completely? if possible I would  like to remove the `app.quarkus.io/commit-id` entriely, and replace the `app.kuberntes.io/name` key with just `name`. Couldn't find a way to achieve this in the documentation.

Thanks in advance.

### PR DESCRIPTION
**Pull request by Sgitario**
Fix https://github.com/quarkusio/quarkus/issues/26587 
Fix https://github.com/quarkusio/quarkus/issues/28297 
Fix https://github.com/quarkusio/quarkus/issues/28078

### CODE REVIEW

**Comment by Sgitario** [comment]
The test failures are unrelated to these changes. Would you mind to take a look @iocanel ?

File: extensions/kubernetes/kind/deployment/src/main/resources/META-INF/services/io.dekorate.ManifestGeneratorFactory [hunk 1]
```diff
@@ -0,0 +1 @@
+io.dekorate.kind.manifest.KindManifestGeneratorFactory
```
Comment (iocanel) [inline extensions/kubernetes/kind/deployment/src/main/resources/META-INF/services/io.dekorate.ManifestGeneratorFactory:1]: I think that this needs to be addressed on dekorate.

**Review by iocanel — APPROVED** [review]
Overall, it looks good to me.

File: extensions/kubernetes/kind/deployment/src/main/resources/META-INF/services/io.dekorate.ManifestGeneratorFactory (hunk 1 above)
Comment (Sgitario) [reply extensions/kubernetes/kind/deployment/src/main/resources/META-INF/services/io.dekorate.ManifestGeneratorFactory:1]: +1

Comment (gsmet) [reply extensions/kubernetes/kind/deployment/src/main/resources/META-INF/services/io.dekorate.ManifestGeneratorFactory:1]: So should we merge this one or is there some additional work planned we want to push to this PR?

Comment (Sgitario) [reply extensions/kubernetes/kind/deployment/src/main/resources/META-INF/services/io.dekorate.ManifestGeneratorFactory:1]: No, the PR is ready to be merged. 
Take into account that this PR depends on Fabric8 Kubernetes Client 6.x which is only in main.

**Pull request merged**